pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -423,6 +423,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		}
 	}
 
+	if pipelineMeta.VerificationResult != nil && pipelineMeta.VerificationResult.VerificationResultType == trustedresources.VerificationError {
```

Comment:
This is to keep the pipelinerun.go behave like before, we have an error check above 
```
	case errors.Is(err, trustedresources.ErrResourceVerificationFailed):
		message := fmt.Sprintf("PipelineRun %s/%s referred pipeline failed signature verification", pr.Namespace, pr.Name)
		pr.Status.MarkFailed(ReasonResourceVerificationFailed, message)
		return controller.NewPermanentError(err)
```
I just realized that I should remove that error check since we now rely on the VerificationResult. 

For the test, yes, we have reconciler tests here: 
https://github.com/tektoncd/pipeline/blob/093517a03e52f357009a660edd7cd968b2da02a9/pkg/reconciler/pipelinerun/pipelinerun_test.go#L11619 

This can make sure if GetPipelineData returns a verificationResult w/ type VerificationError, the pipelinerun is marked as failed. 

I will openother PRs to update  taskrun and pipelinerun code to check the VerificationResult and update conditions.